File: 003/Python/Problem003Tests.py
#!/usr/bin/env python

import unittest

import problem003

class Problem003Tests(unittest.TestCase):

  def testGreatestPrimeFactor(self):
    gpf = problem003.greatestPrimeFactor
    self.assertEqual(2, gpf(2))
    self.assertEqual(2, gpf(4))
    self.assertEqual(3, gpf(6))
    self.assertEqual(29, gpf(13195))

unittest.main()

File: 003/Erlang/problem003_tests.erl
-module(problem003_tests).

-include_lib("eunit/include/eunit.hrl").
-import(problem003, [greatest_prime_factor/1]).

greatest_prime_factor_test_() -> [
  ?_assertEqual(2, greatest_prime_factor(2)),
  ?_assertEqual(2, greatest_prime_factor(4)),
  ?_assertEqual(3, greatest_prime_factor(6)),
  ?_assertEqual(29, greatest_prime_factor(13195))
	].
